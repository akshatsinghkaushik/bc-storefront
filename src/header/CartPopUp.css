.cart-popup {
  position: absolute;
  top: 4.1rem;
  width: 30%;
  height: max-content;
  min-height: max-content;
  background-color: white;
  left: 69.9%;
  justify-content: center;
  flex-direction: column;
  z-index: 100;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.popup-items-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
}

.popup-items {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 95%;
  padding: 0.5rem;
}

.popup-item-details {
  margin: 0.5rem 0 0 1rem;
}

.popup-item-details h3 {
  margin: 0;
  padding: 0;
  font-size: medium;
}

.popup-item-action {
  background: none;
  color: inherit;
  border: none;
  padding: 1rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  transform: translateY(-55%);
}

.popup-product {
  display: flex;
  flex-direction: row;
}

.popup-item-image {
  width: 5rem;
  height: 5rem;
  object-fit: cover;
}

.popup-footer {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.popup-footer-checkout {
  margin: 1rem 1rem 1rem 0;
  background: none;
  background-color: black;
  color: white;
  border: none;
  padding: 1rem 2rem 1rem 2rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  font-size: small;
}

.popup-footer-view {
  margin: 1rem 0 1rem 1rem;
  box-shadow: 0px 0px 0px 1px black inset;
  text-decoration: none;
  background: none;
  background-color: white;
  color: black;
  padding: 1rem 2rem 1rem 2rem;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  font-size: small;
}
