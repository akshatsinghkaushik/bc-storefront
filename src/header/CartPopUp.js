import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import './CartPopUp.css';
import CartContext from '../context/CartContext';

const CartPopUp = ({ cartPopUp, setCartPopUp }) => {
  const cartContext = useContext(CartContext);

  return (
    <div
      className="cart-popup"
      style={cartPopUp ? { display: 'flex' } : { display: 'none' }}
    >
      <div className="popup-items-container">
        {cartContext.cartMapArr.length === 0 ? (
          <div style={{ textAlign: 'center', padding: '1rem' }}>
            No Items in Cart
          </div>
        ) : (
          ''
        )}
        {cartContext.cartMapArr.map((item, index) => {
          return (
            <div key={`popup-item-${index}`} className="popup-items">
              <div className="popup-product">
                <img
                  className="popup-item-image"
                  src={`/media/${item.image}`}
                  alt={item.title}
                />
                <div className="popup-item-details">
                  <h3>
                    {item.title}{' '}
                    <span style={{ fontSize: 'small', fontWeight: 'normal' }}>
                      x{item.quantity}
                    </span>
                  </h3>
                  <p>{item.brand}</p>
                  <p>${Number(item.price * item.quantity).toFixed(2)}</p>
                </div>
              </div>
              <button
                className="popup-item-action"
                onClick={() => cartContext.deleteCartItem(item)}
              >
                X
              </button>
            </div>
          );
        })}
      </div>
      <hr style={{ height: '1px', width: '100%', margin: '0' }} />
      <div className="popup-footer" style={{ margin: '1rem 1rem 0 1rem' }}>
        <p>Total</p>
        <p>${Number(cartContext.getCartTotal()).toFixed(2)}</p>
      </div>
      <div className="popup-footer">
        <Link
          to={'/cart'}
          className="popup-footer-view"
          onClick={() => setCartPopUp(!cartPopUp)}
        >
          VIEW CART
        </Link>
        <button className="popup-footer-checkout">CHECKOUT</button>
      </div>
    </div>
  );
};

export default CartPopUp;
