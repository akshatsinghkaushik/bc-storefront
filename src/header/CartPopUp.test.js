import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CartPopUp from './CartPopUp';
import MockCartProvider from '../tests/MockCartProvider';

it('CartPopUp Renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MockCartProvider>
          <CartPopUp />
        </MockCartProvider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
