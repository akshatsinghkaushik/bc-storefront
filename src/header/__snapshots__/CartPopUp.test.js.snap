// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartPopUp Renders correctly 1`] = `
<div
  className="cart-popup"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="popup-items-container"
  >
    <div
      style={
        Object {
          "padding": "1rem",
          "textAlign": "center",
        }
      }
    >
      No Items in Cart
    </div>
  </div>
  <hr
    style={
      Object {
        "height": "1px",
        "margin": "0",
        "width": "100%",
      }
    }
  />
  <div
    className="popup-footer"
    style={
      Object {
        "margin": "1rem 1rem 0 1rem",
      }
    }
  >
    <p>
      TOTAL
    </p>
    <p>
      $
      0.00
    </p>
  </div>
  <div
    className="popup-footer"
  >
    <a
      className="popup-footer-view"
      href="/cart"
      onClick={[Function]}
    >
      VIEW CART
    </a>
    <button
      className="popup-footer-checkout"
    >
      CHECKOUT
    </button>
  </div>
</div>
`;
